collections/_posts/2023-07-06-Mean_teachers_are_better_role_models.md

```diff
@@ -0,0 +1,202 @@
+- improves the quality of the prediction 
+- better results when those predictions are used as the teacher predictions
+
+Limitation of Temporal ensembling :
+- Each **target** (prediction) is **updated only once per training epoch**.
+- The learned information is **incorporated** into the training process at a relatively **slow pace**.
+- The **larger the dataset**, the **longer it takes** for the updates to span the entire dataset.
+
+&rarr; The authors propose the Mean Teacher model that **averages model weights**, instead of predictions like TE, to improve performances of semi-supervised model.
+
+![](/collections/images/mean_teachers/binary_classification_2.jpg)
+
+A sketch of a binary classification task with two labeled examples (large black dots) and one unlabeled example, demonstrating how the choice of the unlabeled target (blue circle) affects the fitted function (gray curve).
+
+- (a) A model with no regularization &rarr; free to fit any function that predicts the labeled training examples well  
+- (b) A model trained with noisy labeled data (small dots) &rarr; consistent predictions around labeled data points  
+- (c) The teacher model (gray curve) is first fitted to the labeled examples, and then left unchanged during the training of the student model. &rarr; Consistency to noise around unlabeled examples provides additional smoothing  
+- (d) Noise on the teacher model reduces the bias of the targets without additional training. &rarr; The expected direction of stochastic gradient descent is towards the mean (large blue circle) of individual noisy targets (small blue circles)  
+- (e) Ensemble model &rarr; gives an even better expected target
+
+---
+## Mean Teacher 
+
+The Mean Teacher is an average of consecutive student models.
+
+![](/collections/images/mean_teachers/mean_teacher_method.jpg)
+
+- Both the student and the teacher model **evaluate the input applying noise (η, η')**
+- The sotmax output is compared with one-hot label using **classification cost** for the student model and **consistency cost** for the teacher 
+- The teacher model weights are updated using the **EMA weights of the student model**.
+- Both predictions can be used but the **teacher prediction if more likely to be correct**
+- No classification cost is applied when it is a training step with unlabeled example. 
+
+### Consistency cost 
+
+With $$J$$ the consistency cost, as the expected distance between the prediction of the student model (with weights $$θ$$ and noise $$η$$) and the prediction of the teacher model (with weights $$θ'$$ and noise $$η'$$) :
+
+$$ J(θ) = \underset{x,η',η}{E} [ || f (x, θ', η') - f (x, θ, η) ||^2 ] $$
+
+- Mean squared error as the consistency cost and ramp up its weight from 0 to its final value during the first 80 epochs.
```

Comment:
```suggestion
Mean squared error is applied to the distance to get the consistency cost. The weight of this consistency loss changes during training: it ramps up from 0 to its final value during the first 80 epochs.
```

I think it might be a bit more clear this way